client/components/banner/index.jsx

```diff
@@ -128,6 +130,10 @@ export class Banner extends Component {
 			);
 		}
 
+		if ( ! icon ) {
```

Comment:
This would be a breaking change, making all existing `Banner` instances with no defined icon, uh... icon-less. Maybe another approach would be to only call `getIcon()` conditionally, if `props.compact` is false?

Reply:
I was thinking about that, but in the event we wanted to show an icon on the compact style, that wouldn't work.

I'd also considered using a unique string to disable the icon, ie. `no-icon`. That's not very intuitive, but would avoid breakage. What do you think?

There's always the hide-it-with-CSS solution if we wanted to be truly evil. 👿

Reply:
> I'd also considered using a unique string to disable the icon, ie. no-icon. That's not very intuitive, but would avoid breakage. What do you think?

Hm. Maybe I'm struggling with the need to have no icon by default; why is that important?

That said, there only appears to be about 70 instances of `<Banner`, so there's no reason we couldn't go with no icon by default, and just update those existing instances with an explicit icon value.

Reply:
@kwight There's a new prop of `showIcon` which should default to true https://github.com/Automattic/wp-calypso/pull/38733/files#diff-b07c81c57e6e96969fbfc4832226500dR78

See convo in https://github.com/Automattic/wp-calypso/pull/38733#discussion_r366023457 and https://github.com/Automattic/wp-calypso/pull/38733#discussion_r366046006